packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts

```diff
@@ -68,6 +69,234 @@ export enum InstanceInitiatedShutdownBehavior {
+   *
+   * @default Interface type is `InterfaceType.INTERFACE`.
+   */
+  readonly interfaceType?: InterfaceType;
+
+  /**
+   * The number of IPv4 prefixes to be automatically assigned to the network interface. You cannot use this option if you use the Ipv4Prefix option.
+   *
+   * @default One prefix is assigned
+   */
+  readonly ipv4PrefixCount?: number;
+
+  /**
+   * One or more IPv4 prefixes to be assigned to the network interface. You cannot use this option if you use the Ipv4PrefixCount option.
+   *
+   * @default A single prefix is assigned via `ipv4PrefixCount`
+   */
+  readonly ipv4Prefixes?: Ipv4PrefixSpecification[];
+
+  /**
+   * The number of IPv6 addresses to assign to a network interface. Amazon EC2 automatically selects the IPv6 addresses from the subnet range. You can't use this option if specifying specific IPv6 addresses.
+   *
+   * @default No IPv6 addresses are assigned.
+   */
+  readonly ipv6AddressCount?: number;
+
+  /**
+   * One or more specific IPv6 addresses from the IPv6 CIDR block range of your subnet. You can't use this option if you're specifying a number of IPv6 addresses.
+   *
+   * @default No IPv6 addresses are assigned.
+   */
+  readonly ipv6Addresses?: Ipv6Add[];
+
+  /**
+   * The number of IPv6 prefixes to be automatically assigned to the network interface. You cannot use this option if you use the Ipv6Prefix option.
+   *
+   * @default No prefixes are assigned.
+   */
+  readonly ipv6PrefixCount?: number;
+
+  /**
+   * One or more IPv6 prefixes to be assigned to the network interface. You cannot use this option if you use the Ipv6PrefixCount option.
+   *
+   * @default No prefixes are assigned.
+   */
+  readonly ipv6Prefixes?: Ipv6PrefixSpecification[];
+
+  /**
+   * The index of the network card. Some instance types support multiple network cards. The primary network interface must be assigned to network card index 0. The default is network card index 0.
+   *
+   * @default First network interface card gets assigned index `0`.
+   */
+  readonly networkCardIndex?: number;
+
+  /**
+   * The primary IPv6 address of the network interface. When you enable an IPv6 GUA address to be a primary IPv6, the first IPv6 GUA will be made the primary IPv6 address until the instance is terminated or the network interface is detached. For more information about primary IPv6 addresses, see RunInstances.
+   *
+   * @default First IPv6 address is made primary.
+   */
+  readonly primaryIpv6?: boolean;
+
+  /**
+   * The primary private IPv4 address of the network interface.
+   *
+   * @default A random IP address is assigned from one of the subnets.
+   */
+  readonly privateIpAddress?: string;
+
+  /**
+   * One or more private IPv4 addresses.
+   *
+   * @default A random IP address is assigned from one of the subnets.
```

Comment:
I was never able to figure out how these defaults are resolved since the AWS documentation doesn't have any info  
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-privateipadd.html  
```
PrivateIpAddresses

    One or more private IPv4 addresses.

    Required: No

    Type: Array of [PrivateIpAdd](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-privateipadd.html)

    Update requires: [No interruption](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html#update-no-interrupt)
```